src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -411,6 +411,71 @@ public static void ParseJson_UnseekableStream_Async_WithBOM(bool compactData, Te
                     GetAwaiter().GetResult());
         }
 
+        [Fact]
+        public static void ParseJson_Stream_Throws_ClearRentedBuffer_CodeCoverage()
+        {
+            byte[] data = Encoding.UTF8.GetBytes(SR.Json400KB);
+            Stream stream = new ThrowOnReadStream(data);
+            JsonDocument doc = null;
+            Assert.Throws<NotImplementedException>(() =>
+            {
+                doc = JsonDocument.Parse(stream);
+            });
+           Assert.Null(doc);
+        }
+
+        [Fact]
+        public static void ParseJson_Stream_Async_Throws_ClearRentedBuffer_CodeCoverage()
+        {
+            byte[] data = Encoding.UTF8.GetBytes(SR.Json400KB);
+            Stream stream = new ThrowOnReadStream(data);
+            JsonDocument doc = null;
+            Assert.ThrowsAsync<NotImplementedException>(async () =>
+            {
+                doc = await JsonDocument.ParseAsync(stream);
+            });
+            Assert.Null(doc);
+        }
+
+        [Fact]
+        public static void ParseJson_Stream_CanSeekThrows_NullBuffer_CodeCoverage()
+        {
+            byte[] data = Encoding.UTF8.GetBytes(SR.Json400KB);
```

Comment:
If we're expecting the test to throw, do we need such a large object?

Reply:
No need, at first i was disposing the stream while parsing it (for Stream.Read though not for CanSeek), in that case I was needed bigger, will update